plugins/storage/volume/solidfire/src/main/java/org/apache/cloudstack/storage/datastore/driver/SolidFirePrimaryDataStoreDriver.java

```diff
@@ -188,12 +192,25 @@ public boolean grantAccess(DataObject dataObject, Host host, DataStore dataStore
     }
 
     @Override
-    public void revokeAccess(DataObject dataObject, Host host, DataStore dataStore)
-    {
+    public void revokeAccess(DataObject dataObject, Host host, DataStore dataStore) {
         if (dataObject == null || host == null || dataStore == null) {
             return;
         }
 
+        // Workaround: don't unplug iscsi lun when volume is attached to a VM
+        // This is regression workaround from upper layers which are calling
+        // ::releaseVmResources that calls the revoke on attached disk of a VM
+        if (dataObject.getType() == DataObjectType.VOLUME) {
```

Comment:
Explained the workaround in the comments, prior to ScaleIO integration the revokeAccess() wasn't called by upper layers say when a VM is stopped. We don't want to eject/remove the LUN from VMware unless the volume is detached or the VM/volume is to be deleted.

Reply:
can be a separate method

Reply:
Can be done

Reply:
done